test/insertion.bats

```diff
@@ -1,38 +1,111 @@
 load test_helper
 
-@test "insertion target" {
-    result="$(p -a Alamofire -t Altel -f $suts/podfile-altel -d)"
+setup() {
+  init
+}
+
+teardown() {
+  rm -rf $suts/temp
+}
+
+@test "insertion into target" {
+    result="$(p -a Alamofire -t Altel -f $suts/temp -d)"
     text="use_frameworks!
 platform :ios, '10.0'
 inhibit_all_warnings!
 
 target 'Altel' do
 pod 'Alamofire'
   pod 'R.swift'
-  pod 'Sourcery'
   pod 'SwiftFormat/CLI'
-  pod 'SwiftLint'
+end
+
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
 end"
     [ "$result" == "$text" ]
 }
 
-@test "insertion def" {
-   result="$(p -a Alamofire -g insert_pods -f $suts/podfile-gleam -d)"
-   text="def insert_pods
+@test "insertion into: 1) def group" {
+   result="$(p -a Alamofire -g insert_pods -f $suts/temp -d)"
+   text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
+end
+
+def insert_pods
 pod 'Alamofire'
   pod 'Typhoon', '~>4.0’
-  pod 'NVActivityIndicatorView', '~>4.4'
-  pod 'SnapKit', '~>4.2'
-  pod 'RxSwift', '~>4.2'
-  pod 'DTCollectionViewManager', '~> 6.4'
-  pod 'HCSStarRatingView', '~> 1.5'
-  pod 'Moya', '~> 11.0'
+  # Analytics
+end"
+    [ "$result" == "$text" ]
+}
+
+@test "insertion into: 2) comment group" {
+   result="$(p -a Alamofire -g Analytics -f $suts/temp -d)"
+   text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
 end
 
-target 'Gleam' do
-  use_frameworks!
-  # Pods for Gleam
-  insert_pods
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
+  pod 'Alamofire'
 end"
     [ "$result" == "$text" ]
 }
+
+@test "insertion into root" {
+  result="$(p -a Alamofire -p "root" -f $suts/temp -d)"
+  text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+pod 'Alamofire'
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
+end
+
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
+end"
+[ "$result" == "$text" ]
+}
+
+@test "skipping reserved word" {
+  # checking skipping reserved word after all insertions
+  result="$(p -a Alamofire -p 'root' -f $suts/temp)"
+  result="$(p -a Alamofire -t Altel -f $suts/temp)"
+  result="$(p -a Alamofire -g insert_pods -f $suts/temp -d)"
+
+  text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+pod 'Alamofire'
+
+target 'Altel' do
+pod 'Alamofire'
```

Comment:
You need to modify indentation logic so that it would work correctly - and fix tests appropriately.